projects/ccf-eui/src/app/shared/components/spatial-search-config-behavior/spatial-search-config-behavior.component.ts

```diff
@@ -35,17 +34,79 @@ export class SpatialSearchConfigBehaviorComponent {
   @Dispatch()
   readonly updateOrgan = actionAsFn(SetOrgan);
 
+  sex: Sex;
+
+  organ?: OrganInfo;
+
+  panelData: PanelData;
+
+  private readonly subscriptions = new Subscription();
+
+  private readonly dialogSubs = new Subscription();
+
   constructor(
     private readonly dialogRef: MatDialogRef<SpatialSearchConfigComponent>,
-    private readonly spatialSearchDialog: MatDialog
-  ) { }
+    private readonly spatialSearchDialog: MatDialog,
+    public dialog: MatDialog,
+    private readonly infoService: InfoButtonService,
+
+  ) {
+    this.subscriptions.add(this.sex$.subscribe((sex) => {
+      this.sex = sex;
+    }));
+    this.subscriptions.add(this.selectedOrgan$.subscribe((organ) => {
+      this.organ = organ;
+    }));
+  }
 
   buttonClicked(): void {
-    this.spatialSearchDialog.open(SpatialSearchUiBehaviorComponent);
+    this.spatialSearchDialog.open(SpatialSearchUiBehaviorComponent, {
+      data: {
```

Comment:
SpatialSearchUiBehavior will be able to get this from the SpatialSearchUiState, so no need to pass this data over. I should have a PR in this morning you can review. There will be quite a few things to refactor to get in line with it, but it's in the ballpark.